syntax = "proto3";

import "google/protobuf/timestamp.proto";

//add more:
message LocationMessageRequest {
  string id = 1;
  int32 person_id = 2;
  float longitude = 3;
  float latitude = 4;
  google.protobuf.Timestamp creation_time = 5;
}

message LocationMessageResponse {
  string id = 1;
  int32 person_id = 2;
  string coordinate =3;
  string creation_time = 4;
}

message Empty {

}

service LocationService {
  rpc Create(LocationMessageRequest) returns (LocationMessageResponse);
  rpc Retrieve(Empty) returns (LocationMessageResponse);
}
